scripts/main.py

```diff
@@ -345,18 +348,64 @@ def main():
         np.random.seed(args.seed)
         torch.manual_seed(args.seed)
         torch.cuda.manual_seed(args.seed)
+
     # train & resume
     ##########################################################################
     if args.mode == "train":
-        args.files_train = gather_files_from_cmdline(args.files_train)
-        args.val_files = gather_files_from_cmdline(args.val_files)
-        _logger.debug("Training data:   " + "\n".join(args.files_train))
+
+        args.cleanpeakfile = gather_files_from_cmdline(args.cleanpeakfile)
+        args.noisybw = gather_files_from_cmdline(args.noisybw, extension=".bw")
+        args.cleanbw = gather_files_from_cmdline(args.cleanbw, extension=".bw")
+
+        # We have to make sure there is a 1-1 correspondence between files.
+        assert len(args.cleanpeakfile) == len(args.noisybw)
+        assert len(args.cleanbw) == len(args.noisybw)
+
+        train_files = []
```

Comment:
since `train_files` and `val_files` are generated from respective clean/noisy tracks, is there a way to run training/validation from pre generated .h5 files?

Reply:
Right now, API does not allow for that. I felt like it would cause a lot of confusion for users to have a lot of input flexibility. If there is a real cost of time/money involved, we can reconsider this in the future. 
 WDYT ?